app/controllers/teachers_controller.rb

```diff
@@ -55,33 +61,58 @@ def create
   def edit
     @teacher = Teacher.find(params[:id])
     @school = @teacher.school
+    @status = is_admin? ? "Admin" : "Teacher"
+    @action = "Update"
+    @readonly = !is_admin?
   end
 
   def update
     @teacher = Teacher.find(params[:id])
     @school = @teacher.school
+    old_email, old_snap = @teacher.email, @teacher.snap
+    new_email, new_snap = teacher_params[:email], teacher_params[:snap]
+    if ((old_email != new_email) || (old_snap != new_snap)) && !is_admin?
```

Comment:
you should be able to write `@teacher.email_changed?` if you move this logic after `update()`,